src/client/httpClient.ts

```diff
@@ -256,15 +256,27 @@ function deserialize(data: Record<string, any>): Record<string, any> {
 
 function handleValues(key: any, value: any): any {
   if (value != null && typeof value === 'bigint') {
+    //BigInt handling
     return Number(value);
   } else if (value != null && typeof value === 'object') {
     // ObjectId to strings
-    if (value.$oid) return value.$oid;
+    if (value.$oid) {
+      return value.$oid;
+    } else if (value.$numberDecimal) {
+      //Decimal128 handling
+      return Number(value.$numberDecimal);
```

Comment:
I think it may be worthwhile for us to throw an error on Decimal128 rather than just treating Decimal128 as a number? I'm not sure how Cassandra handles floating points, but the purpose of the Decimal128 type is to use decimal floating points, rather than binary floating points. [Here's some notes on the difference](https://thecodebarbarian.com/a-nodejs-perspective-on-mongodb-34-decimal.html).

Unless JavaScript numbers end up as decimal floating points in Cassandra, I think we should throw an error on Decimal types.

Reply:
On Java side (before Cassandra), we actually deal with `BigDecimal` which is decimal-based: all numbers get converted to it, not binary floating-points. For what that's worth.
I think (but may be wrong) that we are also using `decimal` in Cassandra, with similar qualities.

Reply:
@tatu-at-datastax I looked and it looks like you're right, `0.1 + 0.2 === 0.3` with JSON API. In that case, I retract my concerns, we can support Decimal128 in stargate-mongoose :+1:

Reply:
@amorton I think your original decision to go with `BigDecimal` / `decimal` is proving beneficial here :)